Source/com/drew/imaging/png/PngMetadataReader.java

```diff
@@ -166,7 +167,9 @@ private static void processChunk(@NotNull Metadata metadata, @NotNull PngChunk c
                 int bytesLeft = bytes.length - profileNameBytes.length - 2;
                 byte[] compressedProfile = reader.getBytes(bytesLeft);
                 InflaterInputStream inflateStream = new InflaterInputStream(new ByteArrayInputStream(compressedProfile));
-                new IccReader().extract(new RandomAccessStreamReader(inflateStream), metadata, directory);
+                IccDirectory iccdirectory = new IccReader().extract(new RandomAccessStreamReader(inflateStream)); //, metadata, directory);
```

Comment:
As you know, the .NET API doesn't have the `Metadata` object, however C# has Linq which makes working with a collection of `Directory` objects simple. In Java, the `Metadata` object makes things simpler for the user.

There might be an opportunity to return directories instead of passing `Metadata` to readers, but I would vote that we do that for all readers, and possibly in a major version change. It's better if users can upgrade the library without breaking API changes. As far as I can tell this API changes to `IccReader` and `XmpReader` are not strictly necessary.